src/core/misc.c

```diff
@@ -261,6 +261,26 @@ char *gslist_to_string(GSList *list, const char *delimiter)
 	return ret;
 }
 
+gboolean gslist_is_unique(GSList *list)
+{
+	const char *prev;
+
+	if (list == NULL || list->next == NULL)
+		return TRUE;
+
+	prev = (const char *)list->data;
+	list = list->next;
+
+	while (list != NULL) {
+		if (strcmp(list->data, prev) != 0)
+			return FALSE;
+		prev = (const char *)list->data;
+		list = list->next;
+	}
+
+	return FALSE;
```

Comment:
This function only returns `TRUE` iff our list is the singleton list or an empty list?

I think this function should be reconsidered, both the name and the implementation of it. It looks like the function was named initially to do one things, but was turned into something else.

If you want to make a function, that checks that all values in the input list is the same, then you should probably take the comparison function as an argument to the function.

I think this function will also (with the current implementation in mind) require that the input list is always sorted.